tests/gem5/suite_tests/configs/suite_run_workload.py

```diff
@@ -0,0 +1,146 @@
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""
+This example runs the first workload of the given suite.
+
+Characteristics
+---------------
+* User needs to specify the isa in lower case.
+"""
+import argparse
+
+from gem5.components.cachehierarchies.classic.private_l1_private_l2_cache_hierarchy import (
+    PrivateL1PrivateL2CacheHierarchy,
+)
+from gem5.components.memory import SingleChannelDDR3_1600
+from gem5.components.processors.cpu_types import CPUTypes
+from gem5.components.processors.simple_processor import SimpleProcessor
+from gem5.isas import ISA
+from gem5.resources.resource import obtain_resource
+from gem5.simulate.simulator import Simulator
+from gem5.utils.requires import requires
+
+parser = argparse.ArgumentParser(description="A script to run suite tests.")
+
+parser.add_argument(
+    "-i", "--id", type=str, required=True, help="The suite id."
+)
+
+parser.add_argument(
+    "-v", "--version", type=str, required=False, help="The suite version."
+)
+
+parser.add_argument(
+    "-t",
+    "--tick-exit",
+    type=int,
+    required=False,
+    help="The tick to exit the simulation.",
+)
+
+parser.add_argument(
+    "-r",
+    "--resource-directory",
+    type=str,
+    required=False,
+    help="The directory in which resources will be downloaded or exist.",
+)
+
+parser.add_argument(
+    "-s",
+    "--isa",
+    type=str,
+    required=True,
+    help="The ISA to use.",
+)
+
+args = parser.parse_args()
+
+
+# Setup the cache hierarchy.
+cache_hierarchy = PrivateL1PrivateL2CacheHierarchy(
+    l1d_size="32KiB", l1i_size="32KiB", l2_size="512KiB"
+)
+
+# Setup the system memory.
+memory = SingleChannelDDR3_1600(size="3GB")
+
+
+def get_processor(isa):
+    processor = SimpleProcessor(cpu_type=CPUTypes.TIMING, isa=isa, num_cores=1)
+    return processor
+
+
+if args.isa == "riscv":
+    requires(isa_required=ISA.RISCV)
+
+    from gem5.prebuilt.riscvmatched.riscvmatched_board import RISCVMatchedBoard
+
+    board = RISCVMatchedBoard()
+    is_fs = False
+
+if args.isa == "x86":
+    requires(isa_required=ISA.X86)
+
+    from gem5.components.boards.x86_board import X86Board
+
+    processor = get_processor(ISA.X86)
+    board = X86Board(
+        clk_freq="1GHz",
+        processor=processor,
+        memory=memory,
+        cache_hierarchy=cache_hierarchy,
+    )
+    is_fs = True
```

Comment:
Why are all x86 suites full system?

Reply:
The only 2 x86 suites are gapbs and npb which are fs.

Reply:
I think this should be an option and not hardcoded. I don't think it makes sense to hardcode whether the *suite* is fs or not in the run script. This should be information with the suite. In this case, it should be in the test description file.